packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -571,6 +573,7 @@ class HeroController extends NavigatorObserver {
   }
 
   void _updateQuest(Duration timeStamp) {
+    assert(!_questsEnabled);
     if (navigator == null) {
       // The navigator was removed before this end-of-frame callback was called.
       _clearPendingHeroQuest();
```

Comment:
does this branch mean that if go down this branch, you'll never re-enable heroes even if you re-use the object?